### ISSUE TITLE
build:payload fails on some systems

### ISSUE THREAD

**Issue by mryellow** [payloadcms/payload#6611]
### Describe the Bug

We've had issues on Debian, Ubuntu, however a MacOS machine was able to run it.

With the TypeScript I had `Error: unknown field cwd` same as other reports:

https://help.railway.app/questions/i-want-to-deploy-a-nextjs-project-with-p-dc946ee6
https://help.railway.app/questions/error-unknown-field-cwd-when-building-edc1598f
https://github.com/swc-project/swc/discussions/7538#discussion-5306254 

These appear to have been solved by installing C++ Redistributable on Windows machines.

...

Now with a Javascript version of the files I get:

```
> cross-env PAYLOAD_CONFIG_PATH=src/payload.config.js payload build

/home/user/path/node_modules/@swc/wasm/wasm.js:489
    const ret = new Error(getStringFromWasm0(arg0, arg1));
                ^

Error: invalid type: unit value, expected a map
```

### To Reproduce

`npm run build:payload`

### Payload Version

2

**Comment by AlessioGr** [comment]
hey @mryellow can you share your exact payload and node version?

**Comment by mryellow** [comment]
node: `v20.11.1`
payload: `2.18.3`

**Comment by mryellow** [comment]
The Ubuntu server was Node `v16.19.1`

Debian was `v20.11.1`.

Both had the ```Error: unknown field `cwd```` variant of the error with TypeScript app built with CLI tools.

**Comment by mryellow** [comment]
Neither Linux system has `yarn` installed. I notice you use a bunch for running basic commands, perhaps it adds extra environment.

**Comment by mryellow** [comment]
What does `build:payload` do?

The person running it on MacOS says it's not creating any files... 

Do we need to run it at all?

**Comment by mryellow** [comment]
```
nvm use 16
rm -rf node_modules
npm i
PAYLOAD_CONFIG_PATH=src/payload.config.ts npx payload build
```

This gets past the WASM error and into Webpack. Where there is an error which may have been hidden by the other one.

```
...
asset modules 84 bytes (javascript) 4.37 KiB (asset)
  ./node_modules/payload/dist/admin/assets/images/favicon.svg 42 bytes (javascript) 437 bytes (asset) [built] [code generated]
  ./node_modules/payload/dist/admin/assets/images/og-image.png 42 bytes (javascript) 3.94 KiB (asset) [built] [code generated]

ERROR in node:url
Module build failed: UnhandledSchemeError: Reading from "node:url" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
    at /home/user/path/node_modules/webpack/lib/NormalModule.js:918:25
```

**Comment by Jonashr** [comment]
Similar issue here that started today. No issues before that.

payload v-2.19.0

`
 ERROR in node:url
#13 49.95 Module build failed: UnhandledSchemeError: Reading from "node:url" is not handled by plugins (Unhandled scheme).
`

Builds fine locally, though fails with the error when deployed via Docker on Render.

**Comment by dmvdven** [comment]
A few hours ago I started to experience the same issue. In my case the error is triggered when visiting the Admin UI (/admin). My homepage (front-end) somehow works.

Node version: 20.13.1
Payload version: 3.0.0-beta.32
Using Docker via Orbstack

```
Module build failed: UnhandledSchemeError: Reading from "node:url" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
Import trace for requested module:
node:url
./node_modules/fast-uri/index.js
./node_modules/payload/node_modules/ajv/dist/runtime/uri.js
./node_modules/payload/node_modules/ajv/dist/core.js
./node_modules/payload/node_modules/ajv/dist/ajv.js
./node_modules/payload/dist/fields/validations.js
./node_modules/payload/dist/exports/fields/validations.js
./node_modules/@payloadcms/next/dist/views/ForgotPassword/ForgotPasswordForm/index.js
```

**Comment by dmvdven** [comment]
Fixed the error by adding the "ajv" dependency below (version 8.14.0) to my own package.json file. The problem has to do with the newest "ajv" package being broken (version 8.15.0), see for more details: https://github.com/payloadcms/payload/pull/6607.

```
 "dependencies": {
    "ajv": "8.14.0"
 }
```

I use Payload 3.0 so upgrading to 3.0.0-beta40 gives me some errors (which have nothing to do with this problem), but that's why I use the above method for now.

I think the preferred method to fix this error is to simply update your Payload version to: `2.19.0` or `3.0.0-beta.40`.

**Comment by dmvdven** [comment]
Ajv reverted the changes from 8.1.5 in the new version 8.1.6, see: https://github.com/ajv-validator/ajv/issues/2446.

**Comment by aerosurfin** [comment]
The fix provided by @dmvdven worked for me (thanks, Dennis). 

- MacOS 12.7.4
- node v21.4.0 when on metal, `node:21.7.3-bullseye-slim` when containerized 
- working from a clean Payload distribution via `npx create-payload-app@latest` (`blank` template)

**Comment by dorinoltean** [comment]
TLDR Installing `@swc/core@1.4.0`  fixed the issue

I've encountered a similar issue:
```
#0 3.414 > payload build
#0 3.414
#0 13.05 /home/node/node_modules/@swc/wasm/wasm.js:489
#0 13.05     const ret = new Error(getStringFromWasm0(arg0, arg1));
#0 13.05                 ^
#0 13.05
#0 13.05 Error: invalid type: unit value, expected a map
#0 13.05     at module.exports.__wbindgen_error_new (/home/node/node_modules/@swc/wasm/wasm.js:489:17)
#0 13.05     at wasm://wasm/047124f6:wasm-function[9674]:0xccbd1f
#0 13.05     at wasm://wasm/047124f6:wasm-function[11682]:0xd253d7
#0 13.05     at wasm://wasm/047124f6:wasm-function[13505]:0xd4b80e
#0 13.05     at wasm://wasm/047124f6:wasm-function[1028]:0x6902ce
#0 13.05     at wasm://wasm/047124f6:wasm-function[153]:0x204e55
#0 13.05     at wasm://wasm/047124f6:wasm-function[221]:0x2d4a33
#0 13.05     at wasm://wasm/047124f6:wasm-function[284]:0x365618
#0 13.05     at wasm://wasm/047124f6:wasm-function[229]:0x2e91dc
#0 13.05     at wasm://wasm/047124f6:wasm-function[12726]:0xd3f531
#0 13.05
#0 13.05 Node.js v18.19.1
------
failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1
```

The error only happens when building in docker.
I try, with no success:
* `node:20.14.0`
* `"webpack": "^5.91.0"`

investigating `npm list @swc/core` I see that I am using `@swc/core@1.3.107`

Installing `@swc/core@1.4.0`  fixed the issue

**Comment by DanRibbens** [comment]
@AlessioGr should this issue be closed now since you merged https://github.com/payloadcms/payload/pull/6607?

**Comment by mryellow** [comment]
I'd just like to take a moment to thank @DanRibbens for this being a human instead of @githubactions or @stalebot interjecting themselves.

**Comment by AlessioGr** [comment]
Closing this as this should be resolved now. The ajv issue has been addressed in the latest version. As for the swc issue, this should be fixed in https://github.com/payloadcms/payload/pull/6809 by bumping it to the latest version

**Comment by mryellow** [comment]
`npm i @swc/core@1.6.1`

Result: `Module build failed: UnhandledSchemeError: Reading from "node:url" is not handled by plugins (Unhandled scheme).`

`npm i ajv@8.16.0`

Result: Success.
